src/libclient/measurement/ping/ping_linux.cpp

```diff
@@ -252,10 +252,28 @@ Result Ping::result() const
     QVariantList roundTripMs;
     float avg = 0.0;
     QList<float> tmpPingTime = pingTime;
+    quint32 tmpPingsReceived = m_pingsReceived;
+    quint32 tmpPingsSent = m_pingsSent;
 
     // calculate average and fill ping times
     foreach (float val, tmpPingTime)
     {
+        // RTTs greater than receiveTimeout are discarded and don't influence the result
+        if (val >= definition->receiveTimeout)
+        {
+            if (tmpPingsReceived > 0)
+            {
+                tmpPingsReceived--;
+            }
+
+            if (tmpPingsSent > 0)
```

Comment:
This seems wrong, the ping was sent and should count into loss calculation. Removing it from the received count is fine.